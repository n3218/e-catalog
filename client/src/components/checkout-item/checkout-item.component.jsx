import React from "react"
import { connect } from "react-redux"

import "./checkout-item.styles.scss"
import { clearItemFromCart, addItem, removeItem } from "../../redux/cart/cart.actions"

export const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }) => {
  const { imageUrl, name, quantity, price } = cartItem
  return (
    <div className="checkout-item">
      <div className="image-container">
        <img src={imageUrl} alt={name} />
      </div>
      <span className="name">{name}</span>
      <span className="quantity">
        <div id="checkout-item-remove" onClick={() => removeItem(cartItem)} className="arrow">
          &#10094;
        </div>
        <span className="value">{quantity}</span>
        <div id="checkout-item-add" onClick={() => addItem(cartItem)} className="arrow">
          &#10095;
        </div>
      </span>
      <span className="price">${price}</span>
      <div id="checkout-item-clear" onClick={() => clearItem(cartItem)} className="remove-button">
        &#10005;
      </div>
    </div>
  )
}

const mapDispatchToProps = dispatch => ({
  clearItem: item => dispatch(clearItemFromCart(item)),
  addItem: item => dispatch(addItem(item)),
  removeItem: item => dispatch(removeItem(item))
})

export default connect(null, mapDispatchToProps)(CheckoutItem)
