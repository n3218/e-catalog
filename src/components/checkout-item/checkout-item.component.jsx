import React, { useContext } from 'react';
import './checkout-item.styles.scss';
import { CartContext } from '../../providers/cart/cart.provider'


const CheckoutItem = ({ cartItem }) => {
  const { imageUrl, name, quantity, price } = cartItem
  const { addItem, removeItem, clearItemFromCart } = useContext(CartContext)

  return (
    <div className='checkout-item'>
      <div className='image-container'>
        <img src={imageUrl} alt={name} />
      </div>
      <span className='name'>{name}</span>
      <span className='quantity'>
        <div onClick={() => removeItem(cartItem)} className='arrow'>&#10094;</div>
        <span className='value'>{quantity}</span>
        <div onClick={() => addItem(cartItem)} className='arrow'>&#10095;</div>
      </span>
      <span className='price'>${price}</span>
      <div onClick={() => clearItemFromCart(cartItem)} className='remove-button'>&#10005;</div>
    </div>
  )
}


export default CheckoutItem