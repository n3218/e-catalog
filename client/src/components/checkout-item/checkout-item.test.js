import { shallow, mount, render } from "enzyme"
import React from "react"
import { CheckoutItem } from "./checkout-item.component"

describe("CheckoutItem tests", () => {
  let mockProps
  let wrap
  let mockClearItem = jest.fn()
  let mockAddItem = jest.fn()
  let mockRemoveItem = jest.fn()
  let mockCartItem
  let mockDispatch = jest.fn()

  beforeEach(() => {
    mockCartItem = {
      price: "$20",
      imageUrl: "http://imageurl.jpg",
      name: "name of Product",
      quantity: "2"
    }
    mockProps = {
      clearItem: mockClearItem,
      addItem: mockAddItem,
      removeItem: mockRemoveItem,
      dispatch: mockDispatch
    }
    wrap = shallow(<CheckoutItem cartItem={mockCartItem} {...mockProps} />)
  })

  it("expect to render CheckoutItem component", () => {
    expect(wrap).toMatchSnapshot()
  })

  it("shold call removeItem when arrow is clicked", () => {
    wrap.find("[id='checkout-item-remove']").simulate("click")
    expect(mockRemoveItem).toHaveBeenCalled()
  })

  it("shold call addItem when arrow is clicked", () => {
    wrap.find("[id='checkout-item-add']").simulate("click")
    expect(mockAddItem).toHaveBeenCalled()
  })

  it("shold call clearItem when arrow is clicked", () => {
    wrap.find("[id='checkout-item-clear']").simulate("click")
    expect(mockClearItem).toHaveBeenCalled()
  })
})
