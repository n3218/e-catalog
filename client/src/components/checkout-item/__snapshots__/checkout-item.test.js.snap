// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem tests expect to render CheckoutItem component 1`] = `ShallowWrapper {}`;
